File: source/com/barrybecker4/puzzle/tantrix1/model/PathColors.java
// Copyright by Barry G. Becker, 2012. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.puzzle.tantrix1.model;

import java.util.ArrayList;

/**
 * The list of 6 edge colors starting from the right side and going counter clockwise
 *
 * @author Barry Becker
 */
class PathColors extends ArrayList<PathColor> {

      PathColors(PathColor c1, PathColor c2, PathColor c3, PathColor c4, PathColor c5, PathColor c6)  {
          super(6);
          add(c1);
          add(c2);
          add(c3);
          add(c4);
          add(c5);
          add(c6);
      }
}

File: scala-source/com/barrybecker4/puzzle/tantrix/model/PathColors.scala
// Copyright by Barry G. Becker, 2017. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.puzzle.tantrix.model

import com.barrybecker4.puzzle.tantrix.model.PathColor.PathColor


object PathColor extends Enumeration {
  type PathColor = Value
  val RED, GREEN, BLUE, YELLOW, WHITE = Value
}

/* This should have worked, but didn't */
object PathColors {
  def apply(c1: PathColor, c2: PathColor, c3: PathColor, c4: PathColor, c5: PathColor, c6: PathColor) =
    new PathColors(Array(c1, c2, c3, c4, c5, c6))
}

/** The list of 6 edge colors starting from the right side and going counter clockwise. Immutable */
case class PathColors(colors: Array[PathColor]) {

  def apply(i: Int): PathColor = colors(i)
}